### PR TITLE
Correctly save and handle boolean settings in OIDC administration, limit auth provider by default

### PR DESCRIPTION
**Pull request by oliverguenther**
https://community.openproject.org/work_packages/50432
https://community.openproject.org/work_packages/50448

### CODE REVIEW

File: app/services/users/register_user_service.rb [hunk 1]
```diff
@@ -56,6 +57,16 @@ def call
 
     private
 
+    ##
+    # Check whether the associated single sign-on providers
+    # allows for automatic activation of new users
+    def ensure_provider_not_limited!
+      if limited_provider?(user) && Setting::SelfRegistration.disabled?
```
Comment (machisuji) [inline app/services/users/register_user_service.rb:64]: Thanks for adding the specific error message helping users to help themselves.

However, this only covers the case that the global self registration is disabled. As it stands right now your PR allows OIDC/SAML users to register and activate even when the global self registration is set to manual.

The expected behaviour is that the OIDC/SAML registration should follow the global setting when "Limit self registration" is checked. That is a user should be able to register but **not** be activated if the global setting says manual.

**Review by machisuji — CHANGES_REQUESTED** [review]
@oliverguenther Thanks, this does fix the "use_graph_api" setting for the providers. Unchecking it and saving it now results in "use_graph_api" being false as expected.

The limit self registration functionality isn't working as expected anymore, though. Could you please have a look (see my other comment)?

File: app/services/users/register_user_service.rb (hunk 1 above)
Comment (machisuji) [reply app/services/users/register_user_service.rb:64]: I'm not sure why the spec doesn't catch this but I can definitely activate straight away.

image

image

image

image

**Comment by oliverguenther** [comment]
The test did not fail as `activate` was being stubbed so the user never actually changes statuses. This was quite confusing to me, but I've now added explicit tests for all of the different cases. The important bit was this line was missing: https://github.com/opf/openproject/pull/13876/commits/f58c7cda5e7375b3837a451872c2bca54e1d0ce3#diff-a17b7585f4902955b36e61e3ddc85553af1855b7da91c59cce8dc3b8b5db4d1eR119

I've double checked with a keycloak OIDC and saml idP now

**Comment by machisuji** [comment]
> The test did not fail as `activate` was being stubbed so the user never actually changes statuses. This was quite confusing to me, but I've now added explicit tests for all of the different cases. The important bit was this line was missing: [f58c7cd#diff-a17b7585f4902955b36e61e3ddc85553af1855b7da91c59cce8dc3b8b5db4d1eR119](https://github.com/opf/openproject/commit/f58c7cda5e7375b3837a451872c2bca54e1d0ce3#diff-a17b7585f4902955b36e61e3ddc85553af1855b7da91c59cce8dc3b8b5db4d1eR119)
> 
> I've double checked with a keycloak OIDC and saml idP now

Thanks for fixing and expanding the specs! There just one failure where an expected call is done one too many times. Could you please have a look? Once that is green this is good to merge! 👍

**Review by machisuji — APPROVED** [review]
👍

**Pull request merged**